upd/Hamiltonian.py

```diff
@@ -47,7 +48,69 @@ def to_spatial(self, integral: np.ndarray, sym: int, dense: bool):
         :param dense: dense or sparse matrix; default sparse
         :return:
         """
-        pass
+        # FIXME: Eliminate hard-coded number of sites (k)
+        # make it a class attribute: self._k
+        k = 2
+        if not sym in [1, 2, 4, 8]:
+            raise ValueError('Wrong inpput symmetry')
+        
+        #
+        # Assumption: spatial components of alpha and beta spin-orbitals are equivalent
+        #
+        if isinstance(integral, csr_matrix):
+            spatial_int = csr_matrix((k, k))
+            spatial_int = integral[:k, :k]
+        elif isinstance(integral, dict):
+            spatial_int = {}
+            for p in range(k):
+                # U_pppp
+                spatial_int[(p,p,p,p)] = integral[(p, p + k, p, p + k)]
+                for q in range(k):
+                    # Gamma_pqpq, Pairing_ppqq
+                    spatial_int[(p,q,p,q)] = integral[(p,q,p,q)]
+                    spatial_int[(p,p,q,q)] = integral[(p,p,q,q)]
+        else:
+            raise ValueError('Wrong integral input.')
+        
+        #
+        # Symmetries
+        #
+        if isinstance(spatial_int, csr_matrix):
+            if not sym in [1,2]:
+                raise ValueError('Wrong symmetry for 1-body term')
+            if sym == 1:
+                spatial_int = spatial_int + spatial_int.T - diags(spatial_int.diagonal())
+            else:
+                spatial_int = triu(spatial_int)
+        else:
+            if sym == 1:
+                # Add all terms from eightfold permutational symmetry
+                for i in range(k):
+                    for j in range(k):
+                        spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
+                        spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
+                        spatial_int[(j,i,i,j)] = integral[(i,i,j,j)]
+                        spatial_int[(i,j,j,i)] = integral[(i,i,j,j)]                        
+            elif sym == 2:
+                for i in range(k):
+                    for j in range(k):
+                        spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
+            elif sym == 4:
+                for i in range(k):
+                    for j in range(k):
+                        spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
+                        spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
+            else:
```

Comment:
I might be wrong, but it seems like the part 
```
for i in range: 
     for j in range:
```
is duplicating in ternary operators.
would it be less code with the same functionality if we use something like this:

```
for i in range(k):
    for j in range(k):
        if sym == 1:
            spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
            spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
            spatial_int[(j,i,i,j)] = integral[(i,i,j,j)]
            spatial_int[(i,j,j,i)] = integral[(i,i,j,j)]   
        elif sym == 2:
            spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
        elif sym == 4:
            spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
            spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
```
It seems like it should do the same thing, but using less code
@gabrielasd  what do you think about it?

Reply:
@RichRick1 you are right! I'll fix that, thanks :)